lib/mixlib/install.rb

```diff
@@ -100,6 +100,23 @@ def upgrade_available?
     end
 
     #
+    # Returns a Hash containing the platform info options
+    #
+    def self.detect_platform(shell_type = :sh)
```

Comment:
I think we need to expose these to the outside world in two ways:

1. We need to make `detect_platform()` an instance method that updates platform, platform_version and architecture for the `options` of the instance. The usage of it from chef-ingredient (or any other tool) will be like:
```ruby
installer = Mixlib::Install.new({
  product_name: new_resource.product_name,
  channel: new_resource.channel,
  product_version: new_resource.version
}).detect_platform
```
2. We need to have `self.detect_platform_sh` and `self.detect_platform_ps1` which returns the script to run incase a tool wants to do this remotely on a different node.